hooks/hookexecution/execution.go

```diff
@@ -315,28 +316,36 @@ func handleHookMutations[P any](
 	return payload
 }
 
-func handleModuleActivities[P any](hookCode string, activityControl privacy.ActivityControl, payload P) P {
+func handleModuleActivities[P any](hookCode string, activityControl privacy.ActivityControl, payload P, account *config.Account) P {
 	payloadData, ok := any(&payload).(hookstage.RequestUpdater)
 	if !ok {
 		return payload
 	}
 
 	scopeGeneral := privacy.Component{Type: privacy.ComponentTypeGeneral, Name: hookCode}
 	transmitUserFPDActivityAllowed := activityControl.Allow(privacy.ActivityTransmitUserFPD, scopeGeneral, privacy.ActivityRequest{})
+	transmitPreciseGeoActivityAllowed := activityControl.Allow(privacy.ActivityTransmitPreciseGeo, scopeGeneral, privacy.ActivityRequest{})
 
-	if !transmitUserFPDActivityAllowed {
-		// changes need to be applied to new payload and leave original payload unchanged
-		bidderReq := payloadData.GetBidderRequestPayload()
+	if transmitUserFPDActivityAllowed && transmitPreciseGeoActivityAllowed {
+		return payload
+	}
 
-		bidderReqCopy := ortb.CloneBidderReq(bidderReq.BidRequest)
+	// changes need to be applied to new payload and leave original payload unchanged
+	bidderReq := payloadData.GetBidderRequestPayload()
 
-		privacy.ScrubUserFPD(bidderReqCopy)
+	bidderReqCopy := ortb.CloneBidderReq(bidderReq.BidRequest)
 
-		var newPayload = payload
-		var np = any(&newPayload).(hookstage.RequestUpdater)
-		np.SetBidderRequestPayload(bidderReqCopy)
-		return newPayload
+	if !transmitUserFPDActivityAllowed {
+		privacy.ScrubUserFPD(bidderReqCopy)
+	}
+	if !transmitPreciseGeoActivityAllowed {
```

Comment:
Correct, in practice we will always have an account because we call `hookExecutor.SetAccount(account)` in auction and amp_auction. 
In order to add nil check for account, what do we expect to do if for some reason account is nil?
1) skip activity execution: `privacy.ScrubGeoAndDeviceIP(bidderReqCopy, ipConf)` 
2) set default masking values(which also should be already set) and execute activity with default values:
`account.Privacy.IPv6Config.AnonKeepBits = iputil.IPv6DefaultMaskingBitSize` and
`account.Privacy.IPv4Config.AnonKeepBits = iputil.IPv4DefaultMaskingBitSize`

Reply:
I suggest option 2 so that we still scrub given the sensitive nature.

Reply:
Updated